### PR TITLE
Add Terraform check workflow

### PR DESCRIPTION
**Pull request by ryanwholey**
Adds a workflow to format check, validate and lint terraform modules as seen in the [terraform-aws-transit-gateway](https://github.com/TakeScoop/terraform-aws-transit-gateway/blob/master/.github/workflows/lint-check-validate.yaml) repo

Thoughts on the file name? We use lint-check-validate.yaml in terraform-aws-transit-gateway but we should namespace it for its use in the workflow template so I settled on the more generic "check-terraform-module"

### CODE REVIEW

File: workflow-templates/check-terraform-module.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - id: get-module-directories
+        uses: bendrucker/find-terraform-modules@v1
+    outputs:
+      matrix: ${{ steps.get-module-directories.outputs.modules }}
+  check-validate:
+    runs-on: ubuntu-latest
+    needs: find-modules
+    strategy:
+      matrix:
+        package: ${{ fromJson(needs.find-modules.outputs.matrix) }}
```
Comment (bendrucker) [inline workflow-templates/check-terraform-module.yml:18]: should be module, here and in the `lint` job, no "package" in TF

File: workflow-templates/check-terraform-module.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    defaults:
+      run:
+        working-directory: ${{ matrix.package }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: fetch-tflint-config
+        uses: ryanwholey/remote-tflint-config@v2
```
Comment (bendrucker) [inline workflow-templates/check-terraform-module.yml:40]: Let's get this into the `terraform-linters` org before merging this template. Can give you comments on the action itself as part of that move.

File: workflow-templates/check-terraform-module.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          source-repo: TakeScoop/tflint-config
+      - uses: terraform-linters/setup-tflint@v1
+        name: Setup TFLint
+        with:
+          tflint_version: v0.29.0
+      - run: |
+          tflint \
+            --format compact \
+            --config ${{ steps.fetch-tflint-config.outputs.path }}
```
Comment (bendrucker) [inline workflow-templates/check-terraform-module.yml:50]:
```suggestion
            --config "${{ steps.fetch-tflint-config.outputs.path }}"
```

File: workflow-templates/check-terraform-module.properties.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "check-terraform-module",
```
Comment (bendrucker) [inline workflow-templates/check-terraform-module.properties.json:2]: Filename should probably be `check-terraform-modules.yml` given that the monorepo functionality is a key part of the workflow. Also begs the question of whether we should be using this workflow for root modules too, monorepo bits included. If we had a pipeline w/ Terraform applying before a deploy, that probably demands a separate, partially overlapping workflow. But for now I think this makes sense to use even for root modules.

**Review by bendrucker — CHANGES_REQUESTED** [review]

File: workflow-templates/check-terraform-module.yml (hunk 2 above)
Comment (ryanwholey) [reply workflow-templates/check-terraform-module.yml:40]: lmk when and how and i'm down

Comment (bendrucker) [reply workflow-templates/check-terraform-module.yml:40]: Created https://github.com/terraform-linters/tflint-load-config-action/ and invited you. Won't be a transfer per se but you can push there. If you wanna do an init PR I can do a round of review then.

Comment (ryanwholey) [reply workflow-templates/check-terraform-module.yml:40]: perfect! thanks!

File: workflow-templates/check-terraform-module.properties.json (hunk 4 above)
Comment (ryanwholey) [reply workflow-templates/check-terraform-module.properties.json]: When developing this I had tested it with a find-modules action that output subdirectories and the root directory, so I believe it should work for root modules too. If it doesn't we can fix, I think it makes sense to use the same action for both casess

**Review by bendrucker — APPROVED** [review]

**Pull request merged**